client/lib/mixins/close-on-esc/hoc.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import { filter, last } from 'lodash';
+import React, { PureComponent } from 'react';
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'esc-hoc' );
+let components = [];
+
+function startCloseOnEscForComponent( component ) {
+	components.push( component );
+
+	debug( 'start', components.length );
+
+	if ( components.length > 0 ) {
```

Comment:
>Which I believe will not create duplicate listeners since the arguments are always the same?

Hmm, at first I thought this would not be the case, but in my own testing I can confirm that only a single handler would be attached.